#VRML_SIM R2021a utf8
WorldInfo {
  coordinateSystem "NUE"
}
Viewpoint {
  orientation -0.9438570582525466 -0.32051177236306233 -0.08003784956842123 0.5848195937140572
  position -0.12437745274607469 0.3596235878403442 0.6263687964169252
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
Robot {
  translation 0.01 0.038 0
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor 0.045 -0.0176 0
      }
      device [
        RotationalMotor {
          name "motor roda esquerda"
        }
        PositionSensor {
          name "sensor roda esquerda"
        }
      ]
      endPoint Solid {
        translation 0.045004127923461694 -0.01760095436205577 1.3234967960943285e-06
        rotation -0.999999647772517 -0.0006327988091038336 -0.0005513805483559795 5.758821796483726
        children [
          Transform {
            rotation 0 0 1 1.5707996938995747
            children [
              DEF shape_roda Shape {
                appearance PBRAppearance {
                  baseColor 0.8 0.15 0.3
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.02
                }
              }
            ]
          }
        ]
        name "roda direita"
        boundingObject Transform {
          rotation 0 0 1 1.5707996938995747
          children [
            USE shape_roda
          ]
        }
        physics DEF fisica_roda Physics {
          density -1
          mass 0.005
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        axis -1 0 0
        anchor -0.045 -0.0176 0
      }
      device [
        RotationalMotor {
          name "motor roda direita"
        }
        PositionSensor {
          name "sensor roda direita"
        }
      ]
      endPoint Solid {
        translation -0.045 -0.0176 0
        rotation -1 0 0 0
        children [
          Transform {
            rotation 0 0 1 1.5707996938995747
            children [
              USE shape_roda
            ]
          }
        ]
        name "roda esquerda"
        boundingObject Transform {
          rotation 0 0 1 1.5707996938995747
          children [
            USE shape_roda
          ]
        }
        physics USE fisica_roda
      }
    }
    DEF rus_shape Shape {
      appearance PBRAppearance {
        baseColor 0.58 0.4 1
        roughness 1
        metalness 0
      }
      geometry Box {
        size 0.075 0.075 0.075
      }
    }
    DistanceSensor {
      translation 0 0.01 -0.038
      rotation 0 1 0 1.7016996938995748
      children [
        Transform {
          rotation 0 0 1 -1.5708003061004252
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.9 0.9 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ir0"
      lookupTable [
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
    }
    DistanceSensor {
      translation 0 0.01 0.038
      rotation 0 1 0 -1.5707996938995747
      name "ir1"
      lookupTable [
        0.03 1024 0
        0.2 0 0
      ]
      numberOfRays 2
    }
    GPS {
    }
    Solid {
      translation 0 -0.01 -0.047
      children [
        Transform {
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.4 1 0.2
                roughness 1
                metalness 0
              }
              geometry Box {
                size 0.078 0.01 0.02
              }
            }
          ]
        }
      ]
    }
    DistanceSensor {
      translation 0 -0.015 -0.048
      rotation -0.3309158326386369 0.33435077301850913 0.8824422203696982 -1.7016996938995748
      children [
        Transform {
          rotation 0 0 1 -1.5708003061004252
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 1 0 0
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.01
              }
            }
          ]
        }
      ]
      name "ir3"
      lookupTable [
        0.003 1023 0
        0.1 0 0
      ]
      type "infra-red"
    }
    Camera {
      translation 0 0.037 -0.034
      rotation 0.997247439051535 -0.029256912880747554 -0.06812912999476152 0
      children [
        Transform {
          children [
            DEF shape_cam Shape {
              appearance PBRAppearance {
                baseColor 0.9 0.9 0.1
                roughness 1
                metalness 0
              }
              geometry Sphere {
                radius 0.01
                subdivision 4
              }
            }
          ]
        }
      ]
      boundingObject USE shape_cam
      fieldOfView 1.047
      width 256
      height 128
    }
  ]
  name "rus"
  boundingObject DEF rus_shape Shape {
    appearance PBRAppearance {
      baseColor 0.58 0.4 1
      roughness 1
      metalness 0
    }
    geometry Box {
      size 0.075 0.075 0.075
    }
  }
  physics Physics {
    density -1
    mass 0.2
  }
  controller "meuRobo"
  synchronization FALSE
}
Solid {
  translation 0.18 0.0176 -0.32
  children [
    DEF t_parede Transform {
      children [
        Solid {
          children [
            DEF parede Shape {
              appearance PBRAppearance {
              }
              geometry Box {
                size 0.3 0.1 0.01
              }
            }
          ]
        }
      ]
    }
  ]
  boundingObject USE parede
}
Solid {
  translation 0 0 0.2
  children [
    Transform {
      children [
        USE parede
      ]
    }
  ]
  name "solid(1)"
  boundingObject USE parede
}
Solid {
  translation 0 0 -0.1
  children [
    Transform {
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0 0 0
            roughness 1
            metalness 0
          }
          geometry Box {
            size 0.2 0.001 0.02
          }
        }
      ]
    }
  ]
  name "solid(2)"
}
